### DISCUSSION TITLE
Handling multiple devices of same name/type

### DISCUSSION THREAD

**Discussion by Snypes88** [hcpy2-0/hcpy#102] [Q&A]
I have a steam oven and normal oven at home, and the devices.json identifies both with name oven. I manually changed the name to Oven and SteamOven and these now show up as seperate devices within Home Assistant, however, is this the correct prorcedure? 

<img width="317" alt="image" src="https://github.com/user-attachments/assets/3392b433-e09f-452f-83ea-c06c267a7d26">

When I attempt to post to the steam oven (for bread baking) I publish an MQTT message in HA to the docker container running hcpy I unfortunately get no reesponse. I am checking that the topics/uid I am posting to are writeable however, nothing. This is a screenshot of my MQTT window. 

<img width="391" alt="image" src="https://github.com/user-attachments/assets/b13ba882-791a-4172-858a-c9be01511358">

I'm not sure if the reason is the HADiscovery.py file might be missing the topics I am trying to set, or if the reason could be faulty communication between my MQTT brooker running in my HA instance and the hcpy docker container running on another raspi in my home network... but any help or pointers would be much appreciated as I can now read all the sensor states for both ovens and this solution is muuuuuch more robust than the HA homeconnect integration which I currently use and would like to replace with a local solution.

**Comment by saveriol** [comment]
which version of hcpy are you running? the latest?

Anyway, can you post the relevant 524 section of devices.json?
I don't think "payload_off" is the right value to send. 
If it's the same as my 524, this is the relevant section:

````
            "524": {
                "name": "BSH.Common.Setting.ChildLock",
                "access": "none",
                "available": "false",
                "initValue": "false",
                "refCID": "01",
                "refDID": "00",
                "discovery": {
                    "component_type": "binary_sensor",
                    "payload_values": {
                        "device_class": "lock",
                        "payload_on": false,
                        "payload_off": true
                    }
                }
            },
````

as you can see, there is no write access to this property. ("available": "false")

**Reply by Snypes88** [reply]
This is what's included in my devices.json

            "524": {
                "name": "BSH.Common.Setting.ChildLock",
                "access": "readWrite",
                "available": "true",
                "refCID": "01",
                "refDID": "00",
                "discovery": {
                    "component_type": "binary_sensor",
                    "payload_values": {
                        "device_class": "lock",
                        "payload_on": false,
                        "payload_off": true
                    }
                }
            },

imho it should allow me to read and write to that entity... I chose it because it is one of the few that are also included in HADevices.py if I'm not mistaken in case my issue was that the programs I would like to access are not covered there

**Reply by Snypes88** [reply]
What I am really looking to try is to publish

homeconnect/SteamOven/activeProgram
{"program":8254}

to start the oven with the Hot Air Steam 60% program, then with follow up MQTT publish commands set the deisred temperature and cooking time... this is how I currently do it with the HomeConnect integration in HA, however, their API is unstable and half the time it gets blocked and the oven doesn't realize that it has to then change program further down in the baking program (only hot air for example and then program time 20 minutes for example) leading to half baked loaves of Sourdough 😓

            "8208": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAir"
            },
            "8250": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAir100Steam"
            },
            "8255": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAir30Steam"
            },
            "8254": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAir60Steam"
            },
            "8253": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAir80Steam"
            },
            "8212": {
                "name": "Cooking.Oven.Program.HeatingMode.HotAirGrilling"
            },

**Reply by saveriol** [reply]
It should work.
Question: is OperationalState sensor in "Ready" state when you're trying to run the program?

**Reply by saveriol** [reply]
> This is what's included in my devices.json
> 
> ```
>         "524": {
>             "name": "BSH.Common.Setting.ChildLock",
>             "access": "readWrite",
>             "available": "true",
>             "refCID": "01",
>             "refDID": "00",
>             "discovery": {
>                 "component_type": "binary_sensor",
>                 "payload_values": {
>                     "device_class": "lock",
>                     "payload_on": false,
>                     "payload_off": true
>                 }
>             }
>         },
> ```
> 
> imho it should allow me to read and write to that entity... I chose it because it is one of the few that are also included in HADevices.py if I'm not mistaken in case my issue was that the programs I would like to access are not covered there

in this case you should set value to `true` or `false` and not "payload_on" and "payload_off"

**Reply by Snypes88** [reply]
Thanks for the quick response! I tried to send both true and false, and both as a string and boolean and both don't seem to change the state of the Child lock on the appliance... 

<img width="459" alt="image" src="https://github.com/user-attachments/assets/5e48aa23-c2e6-4bbf-9cac-c09a717930a0">

<img width="364" alt="image" src="https://github.com/user-attachments/assets/9491884f-255a-4de7-a1b8-1804f3978368">

**Reply by saveriol** [reply]
I have another 524 with readWrite access, and this works:

````
action: mqtt.publish
data:
  evaluate_payload: false
  qos: "0"
  retain: false
  topic: homeconnect/hob/set
  payload: '{"uid":524, "value":true}'
````
![image](https://github.com/user-attachments/assets/3bdd7878-96a8-4348-97bd-d225b01656da)

**Comment by Snypes88** [comment]
Thanks for sharing your exact publish command it seems I wasn't always using the right syntax. I believe I've pinpointed the challenges I'm facing with the backend being disconnected on this particular device, as with another homeconnect device on my network I was able to submit multiple requests that were accepted. Any idea how I can force the backend to become connected?

<img width="314" alt="image" src="https://github.com/user-attachments/assets/0dfd457c-9944-40de-a1b5-8911464a8811">
